python/tests/integration/arcticdb/test_update.py

```diff
@@ -0,0 +1,564 @@
+    update_result = lib.update_batch([update], prune_previous_versions=True)
+    assert update_result[0].version == 1    
+
+
+def dataframe_simulate_arcticdb_update_dynamic(expected_df: pd.DataFrame, update_df: pd.DataFrame) -> pd.DataFrame:
+    """ Simulates partially dynamic update, when update df has new columns.
+    The scenario where there are type promotions of existing columns is not covered yet
+    """
+    def add_missing_ending_columns(to_df: pd.DataFrame, from_df: pd.DataFrame):
+        """
+        Modifies `to_df` adding all missing columns at from `from_df` at the end.
+        Note that it checks first that first common column of both are same
+        """
+        assert len(from_df.columns) >= len(to_df.columns)
+        for index, col in enumerate(from_df.columns):
+            if index < len(to_df.columns):
+                message = f"Column [{index}] does not match from_df{from_df.columns[index]} != to_df{from_df.columns[index]}"
+                assert from_df.columns[index] == to_df.columns[index], message
+            else:
+                dtype = from_df[col].dtype
+                if np.issubdtype(dtype, np.integer):
+                    to_df[col] = np.zeros(len(to_df), dtype=dtype)
+                elif np.issubdtype(dtype, np.floating):
+                    to_df[col] = np.full(len(to_df), np.nan, dtype=dtype)
+                elif np.issubdtype(dtype, np.bool_):
+                    to_df[col] = np.full(len(to_df), False, dtype=dtype)
+                elif dtype == object or pd.api.types.is_string_dtype(dtype):
+                    to_df[col] = [None] * len(to_df)
+    add_missing_ending_columns(expected_df, update_df)
+    return dataframe_simulate_arcticdb_update_static(expected_df, update_df)
+
+
+@pytest.mark.storage
+@pytest.mark.parametrize("custom_library", [
+            {'library_options': LibraryOptions(dynamic_schema=True)}
+        ], indirect=True)
+@pytest.mark.only_fixture_params(["lmdb", "real_s3", "real_gcp"])
+def test_update_batch_different_updates_dynamic_schema(custom_library):
+    """ The test examines different types of updates depending on their
+    UpdatePositionType over original dataframe. All updates have additional
+    columns requiring use of dynamic schema. The updates are having also different 
+    sizes compared to original dataframe. Only batch operations are used here
+    """
+    lib: Library = custom_library
+    seed(SEED)
+    start_time = TimestampNumber.from_timestamp(timestamp=Timestamp("1/2/2017"))
+    g = BasicDataFrameGenerator()
+    ug = UpdatesGenerator(g)
+    original_num_cols = 30
+    original_number_rows = 50
+    symbol_prefix = "different types of updates"
+
+    symbol_names = []
+    update_batch:List[UpdatePayload] = []
+    write_batch:List[UpdatePayload] = []
+    expected_results = dict()
+    dataframes_lenghts = [1, original_number_rows // 2, original_number_rows, original_number_rows * 1.2]
```

Comment:
Is column reordering handled in any test here or in the existing unit tests. E.g.
```python
lib.write(pd.DataFrame({"a": [1], "b": [1]}))
lib.write(pd.DataFrame({"b": [1], "a": [1]}))
```
it would be good to test this property of dynamic schema if it's not already.